### ISSUE TITLE
InputDecorator lacks support for material theming

### ISSUE THREAD

**Issue by rolurq** [flutter/flutter#18478]
When using a material input widget, and adding an icon to it, whether it is suffix, prefix or the leading icon, there's no way of showing that icon with a custom icon theme, because in the code the color is fixed:
```dart
  Color _getDefaultIconColor(ThemeData themeData) {
    if (!decoration.enabled)
      return themeData.disabledColor;

    switch (themeData.brightness) {
      case Brightness.dark:
        return Colors.white70;
      case Brightness.light:
        return Colors.black45;
      default:
        return themeData.iconTheme.color;
    }
  }
```
This comes from [input_decorator.dart](https://github.com/flutter/flutter/blob/48f4ff6ddab5520a6a761f4e6079c64000c16d29/packages/flutter/lib/src/material/input_decorator.dart#L1534)
So, if the theme brightness is dark or light (and it can not be of any other value), the color of the icon on inactive state of the input are always `Colors.white70` or `Colors.black45`, and the default case in the switch will never be taken.
There's a solution for this, which is setting the color in the `Icon` widget directly, but that makes the icon have the same color always, so, to have it changed to primary color when the input gets focused I would have to listen to the focus node of the input and change it from outside, but I think there's no need for this, mainly because it will rerender the input decorator from the state of the input parent widget when it can be changed inside the input making the parent stateless.

**Comment by HansMuller** [comment]
You're right that hard wiring the icon colors this way defeats light/dark theming. 

Currently [InputDecoratorTheme](https://docs.flutter.io/flutter/material/InputDecorationTheme-class.html) doesn't include icon colors. It probably should, and the colors would need to depend on the overall theme's brightness.

**Comment by dbecher-ito** [comment]
I came across the same bug and started to create a GitHub issue, as I realized there already is one... there should definitely be a clean way to properly theme the default icon color, as hardcoding it via `Icon(color: ...)` seems fine, but is not as convenient as styling it properly through `IconTheme`. 👍

**Comment by awhitford** [comment]
See related #48882.

I am personally interested in being able to control the default colors for Input Decorations -- the default `black45` is an awful gray with insufficient contrast with my chosen surface color.  Why can we presently only configure the color for when it has _focus_?

Also note that my complaint is not just about the icon color, but the Input Decoration in general -- the icon, label, underline -- they all have the same gray that should be configurable.
